default/AI/ShipDesignAI.py

```diff
@@ -0,0 +1,1602 @@
+"""
+This module deals with the autonomous shipdesign from the AI. The design process is class-based.
+The basic functionality is defined in the class AIShipDesign. The more specialised classes mostly 
+implement the rating function and some additional information for the optimizing algorithms to improve performance. 
+
+Example usage of this module:
+import ShipDesignAI
+myDesign = ShipDesignAI.AIShipDesign_Military()
+myDesign.additional_specifications.enemy_mine_dmg = 10
+best_military_designs = myDesign.optimize_design()  # best designs per planet: (rating,planetID,design_id,cost) tuples
+
+
+Available ship classes:
+- AIShipDesign_Military: basic military ship
```

Comment:
I generally agree with keeping true to naming conventions. I guess I simply am not too fond of PEP8 in this regard. As physicist it seems pretty natural to me to add _additionalInformation behind a camelCase name (e.g. units for variables describing physical quantities). I guess the average variable name in complex physical simulations simply tends to be a lot longer than what PEP8 convention was intended for. I do understand of course we want to stick to PEP8 strictly.

One thing I miss so far about the suggested names is that the class hierarchy is somewhat lost in the name. I feel like this is something that allows easier use 
Simply omitting the underscores doesn't make me too happy. I guess, I will have to think about the issue a little bit more.